Add more tomorrowio sensors

dirrgang opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Breaking change
<!--
  If your PR contains a breaking change for existing users, it is important
  to tell them what breaks, how to make it work again and why we did this.
  This piece of text is published with the release notes, so it helps if you
  write it towards our users, not us.
  Note: Remove this section if this PR is NOT a breaking change.
-->


## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
I added all available core layer sensors provided by the API.

![grafik](https://github.com/home-assistant/core/assets/29854606/41a54564-cf41-420c-8ef0-7b34ed9e53e8)
![grafik](https://github.com/home-assistant/core/assets/29854606/b200db77-e0ea-4c95-afad-2f0845519a02)


## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [X] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [X] The code change is tested and works locally.
- [X] Local tests pass. **Your PR cannot be merged unless tests pass**
- [X] There is no commented out code in this PR.
- [X] I have followed the [development checklist][dev-checklist]
- [X] I have followed the [perfect PR recommendations][perfect-pr]
- [X] The code has been formatted using Black (`black --fast homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.
- [ ] Untested files have been added to `.coveragerc`.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

dirrgang committed:
Added device class to speedtestdotnet sensor entities. (#100500)

Added device class to sensor entities.

cdce8p committed:
Update numpy to 1.26.0 (#100512)

tronikos committed:
Bump opower to 0.0.34 (#100501)

scop committed:
Fix huawei_lte current month up/download sensor error on delete (#100506)

Deleting one of them prematurely deleted the last reset item
subscription that is shared between the two.

frimtec committed:
Fix broken reconnect capability of fritzbox_callmonitor (#100526)

starkillerOG committed:
Try Reolink ONVIF long polling if ONVIF push not supported (#100375)

cdce8p committed:
Fix full black run condition [ci] (#100532)

bieniu committed:
Add `event` platform for Shelly gen2 devices (#99659)

* Add event platform for gen2 devices

* Add tests

* Add removal condition

* Simplify RpcEventDescription; fix availability

* Improve names and docstrings

* Improve the event entity name

* Use async_on_remove()

* Improve tests coverage

* Improve tests coverage

* Prefix the entity name with the device name in the old way

* Black

* Use DeviceInfo object

gwww committed:
Bump elkm1-lib to 2.2.6 (#100537)

cdce8p committed:
Remove duplicate mobile_app client fixture (#100530)

scop committed:
Fix google invalid token expiry test init for UTC offsets > 0 (#100533)

```
$ python3 -q
>>> import datetime, time
>>> time.tzname
('EET', 'EEST')
>>> datetime.datetime.max.timestamp()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: year 10000 is out of range
```

steffenrapp committed:
Add missing conversation service translation (#100308)

* Update services.yaml

* Update strings.json

* Update services.yaml

* Update strings.json

* Update strings.json

* fix translation keys

* Fix translation keys

jpbede committed:
Move co2signal coordinator to its own file (#100541)

* Move co2signal coordinator to its own file

* Fix import

cdce8p committed:
Update typing-extensions to 4.8.0 (#100545)

cdce8p committed:
Update pytest warnings filter (#100546)

janiversen committed:
Adjust codeowners in modbus (#100474)

joostlek committed:
Clean up lyric sensor platform (#100495)

* Clean up lyric sensor platform

* Clean up lyric sensor platform

* Clean up lyric sensor platform

* Update homeassistant/components/lyric/sensor.py

Co-authored-by: Aidan Timson <aidan@timmo.dev>

* Update homeassistant/components/lyric/sensor.py

Co-authored-by: Aidan Timson <aidan@timmo.dev>

* Update homeassistant/components/lyric/sensor.py

Co-authored-by: Aidan Timson <aidan@timmo.dev>

* Update homeassistant/components/lyric/sensor.py

Co-authored-by: Aidan Timson <aidan@timmo.dev>

---------

Co-authored-by: Aidan Timson <aidan@timmo.dev>

jpbede committed:
Set co2signal integration type to service (#100543)

bdraco committed:
Drop codeowner for Magic Home/flux_led (#100557)

bdraco committed:
Drop codeowner for LIFX (#100556)

rappenze committed:
Fibaro streamline hass.data entry (#100547)

* Fibaro streamline hass.data entry

* Fix tests

cdce8p committed:
Add pylint plugin to check for calls to base implementation (#100432)

chemelli74 committed:
Add Vodafone Station sensor platform (#99948)

* Vodafone: add sensor platform

* fix for model VOX30 v1

* fix, cleanup, 2 new sensors

* apply review comments

* apply last review comment

jpbede committed:
Move uptimerobot coordinator to its own file (#100558)

* Move uptimerobot coordinator to its own file

* Fix import of coordinator in platforms

joostlek committed:
Fix lyric feedback (#100586)

joostlek committed:
Change Hue zigbee connectivity sensor into an enum (#98632)

jpbede committed:
Remove platform const in co2signal coordinator (#100592)

elmurato committed:
Add DataUpdateCoordinator to Minecraft Server (#100075)

Lash-L committed:
Fix Roborock send command service calling not being enum (#100574)

joostlek committed:
Add Spotify code owner (#100597)

ViViDboarder committed:
Add config-flow to NextBus (#92149)

rytilahti committed:
Fix xiaomi_miio button platform regression (#100527)

raman325 committed:
Adjust hassfest.manifest based on config.action (#100577)

joostlek committed:
Call async added to hass super in Livisi (#100446)

bdraco committed:
Update zeroconf discovery to use IPAddress objects to avoid conversions (#100567)

bieniu committed:
Add missing translations for Shelly event type states (#100608)

Add missing translations for event type

elmurato committed:
Clean-up Minecraft Server tests (#100615)

Remove patching of getmac, fix typo

bdraco committed:
Use is for UNDEFINED check in async_update_entry (#100599)

bdraco committed:
Switch config flows use newer zeroconf methods to check IP Addresses (#100568)

abmantis committed:
Add integration for IKEA Idasen Desk (#99173)

Co-authored-by: J. Nick Koston <nick@koston.org>

jpbede committed:
Move renson coordinator to its own file (#100610)

jpbede committed:
Move faa_delays coordinator to its own file (#100548)

bdraco committed:
Update enphase_envoy zeroconf checks to use stdlib ipaddress methods (#100624)

bdraco committed:
Update dhcp to use stdlib ipaddress methods (#100625)

bdraco committed:
Avoid double lookups with data_entry_flow indices (#100627)

robinlee09201 committed:
Fix ecobee aux_heat_off always returns to HEAT (#100630)

bdraco committed:
Bump dbus-fast to 2.9.0 (#100638)

bdraco committed:
Bump SQLAlchemy to 2.0.21 (#99745)

Lash-L committed:
Add second test device for Roborock (#100565)

bdraco committed:
Fix hkid matching in homekit_controller when zeroconf value is not upper case (#100641)

emontnemery committed:
Move attributes to be excluded from recording to entity classes (#100239)

Co-authored-by: J. Nick Koston <nick@koston.org>

bdraco committed:
Update doorbird zeroconf checks to use stdlib ipaddress methods (#100623)

bdraco committed:
Avoid ConfigEntry lookups in hass.config_entries.async_entries for domain index (#100598)

unDemian committed:
Bump ismartgate to 5.0.1 (#100636)

dependabot[bot] committed:
Bump tibdex/github-app-token from 2.0.0 to 2.1.0 (#100632)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Lash-L committed:
Bump python-roborock to 0.34.1 (#100652)

bump to 34.1

aman207 committed:
Add Enmax virtual integration to Opower (#100503)

* add enmax virtual integration supported by opower

* update integrations.json

raman325 committed:
Bump zwave-js-server-python to 0.51.3 (#100665)

elmurato committed:
Clean-up Minecraft Server constants (#100666)

bdraco committed:
Avoid calling extract_stack in system_log since it does blocking I/O (#100455)

jpbede committed:
Add test to london underground (#100562)

Co-authored-by: Robert Resch <robert@resch.dev>

mike391 committed:
Add support for Levoit Vital200s purifier (#100613)

edenhaus committed:
Fix missspelled package names (#100670)

FletcherAU committed:
Add Slack thread/reply support (#93384)

jbouwh committed:
Avoid redundant calls to `async_ha_write_state` in MQTT (binary) sensor (#100438)

* Only call `async_ha_write_state` on changes.

* Make helper class

* Use UndefinedType

* Remove del

* Integrate monitor into MqttEntity

* Track extra state attributes and availability

* Add `__slots__`

* Add monitor to MqttAttributes and MqttAvailability

* Write out loop

* Add test

* Make common test and parameterize

* Add test for last_reset attribute

* MqttMonitorEntity base class

* Rename attr and update docstr `track` method.

* correction doct

* Implement as a decorator

* Move tracking functions into decorator

* Rename decorator

* Follow up comment

emontnemery committed:
Move definition of attributes excluded from history to entity classes (#100430)

* Move definition of attributes excluded from history to entity classes

* Revert change which should be in a follow-up PR

* Fix sun unrecorded attributes

* Fix input_select unrecorded attributes

pjanuario committed:
Add ecoforest integration (#100647)

* Add ecoforest integration

* fix file title

* remove host default from schema, hints will be given in the documentation

* moved input validation to async_step_user

* ensure we can receive device data while doing entry setup

* remove unecessary check before unique id is set

* added shorter syntax for async create entry

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* use variable to set unique id

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* Use _attr_has_entity_name from base entity

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* remove unecessary comments in coordinator

* use shorthand for device information

* remove empty objects from manifest

* remove unecessary flag

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* use _async_abort_entries_match to ensure device is not duplicated

* remove unecessary host attr

* fixed coordinator host attr to be used by entities to identify device

* remove unecessary assert

* use default device class temperature trasnlation key

* reuse base entity description

* use device serial number as identifier

* remove unused code

* Improve logging message

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* Remove unused errors

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* Raise a generic update failed

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* use coordinator directly

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* No need to check for serial number

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* rename variable

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* use renamed variable

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* improve assertion

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* use serial number in entity unique id

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* raise config entry not ready on setup when error in connection

* improve test readability

* Improve python syntax

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* abort when device already configured with same serial number

* improve tests

* fix test name

* use coordinator data

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* improve asserts

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* fix ci

* improve error handling

---------

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

jpbede committed:
Move coolmaster coordinator to its own file (#100425)

Lash-L committed:
Add water shortage binary sensor (#100662)

Lash-L committed:
Add last clean sensors to Roborock (#100661)

* Add water shortage binary sensor

* add last clean sensors

* fix tests

* fix tests again

* remove accidentally added binary sensor

c0ffeeca7 committed:
ZHA multiprotocol detected - fix typo (#100683)

emontnemery committed:
Remove async_process_integration_platform_for_component (#100680)

jimmyd-be committed:
Add Renson button entity (#99494)

Co-authored-by: Robert Resch <robert@resch.dev>

bouwew committed:
Update plugwise to v0.33.0 (#100689)

c0ffeeca7 committed:
ZHA multiprotocol detected message: add info (#100686)

bouwew committed:
Add missing step-differentiation for the Plugwise temperature_offset (#100654)

pjanuario committed:
Add number platform to ecoforest (#100694)

* add power number entity to ecoforest integration

* fix number.py header

* minor fixes

* change power to power level

* update comment for native value prop

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* exclude number.py from coverage

---------

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

pjanuario committed:
Add additional sensors to ecoforest integration (#100681)

* add ecoforest additional sensors

* add ecoforest additional sensors

* use StateType

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* use StateType

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* update cpu temp translation

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* use common translation

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* use common on translation

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* use common standby translation

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* update strings

* update strings

* import state type

* relabel preheating

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* add cpu temp disable by default

---------

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

bdraco committed:
Avoid polling in sun sensor entities (#100693)

jbouwh committed:
Avoid redundant calls to `async_write_ha_state` in MQTT mqtt alarm_control_panel (#100691)

Limit state writes for mqtt alarm_control_panel

jbouwh committed:
Avoid redundant calls to `async_write_ha_state` in MQTT climate & water_heater (#100696)

Limit state writes for mqtt climate & water_heater

jbouwh committed:
Assign color_mode for mqtt light as ColorMode (#100709)

pjanuario committed:
Add switch platform to ecoforest integration (#100708)

* add switch platform to ecoforest integration

* ignore switch.py from coverage

* rename mixin

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* update translations

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* remove translation key

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* move attr name to description

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* fix broken change

* use entity description action

* use lambda with awaitable

---------

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

joostlek committed:
Move samsung tv device class outside of constructor (#100712)

jbouwh committed:
Fix mqtt light rgbww update without state topic (#100707)

* Fix mqtt light rgbww update without state topic

* Add @callback decprator and correct mired conv

jbouwh committed:
Avoid redundant calls to `async_write_ha_state` in MQTT light (#100690)

* Limit state writes for mqtt light

* Additional tests and review follow up

DanielWeeber committed:
Add device class to denonavr (#100711)

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

jbouwh committed:
Use shorthand attrs for MQTT cover (#100710)

* User shorthand attrs for MQTT cover

* Cleanup constructor

* Cleanup constructor

starkillerOG committed:
Reolink remove unneeded str() (#100718)

Olen committed:
Add more august actions (#100667)

joostlek committed:
Remove obsolete methods in HVV departures (#100451)

* Call async added to hass super in HVV departures

* Remove obsolete methods

chemelli74 committed:
Bump aiovodafone to 0.3.0 (#100729)

Olen committed:
Only get state once for all August sensor-tests (#100721)

chemelli74 committed:
Bump aiocomelit to 0.0.8 (#100714)

* Bump aiocomelit to 0.0.8

* fix import

* fix tests

jbouwh committed:
Avoid redundant calls to async_write_ha_state in mqtt cover (#100720)

Avoid redundant calls to async_write_ha_state

rappenze committed:
Remove unneeded instance check (#100736)

synesthesiam committed:
Bump intents to 2023.9.22 (#100737)

abmantis committed:
Improve Idasen Desk "no devices found" message (#100742)

dieselrabbit committed:
Bump screenlogicpy to v0.9.1 (#100744)

puddly committed:
Bump ZHA dependencies (#100732)

thecode committed:
Fix Shelly Gen2 event get input name method (#100733)

emontnemery committed:
Fix handling of unit system change in sensor (#100715)

gjohansson-ST committed:
Fix weather template forecast attributes (#100748)

dirrgang committed:
Added all core sensors and extended forecast.

dirrgang committed:
Added all core sensors and extended forecast.

bieniu committed:
Add `event` platform for Shelly gen1 devices (#100655)

* Initial commit

* Use description.key

* Add translations

* Check event_types

* Rename input_id to channel

* Fix removeal confition

* Add tests

* Sort classes and consts

* Use ShellyBlockEntity class

* Update tests

* Update homeassistant/components/shelly/event.py

Co-authored-by: Shay Levy <levyshay1@gmail.com>

---------

Co-authored-by: Shay Levy <levyshay1@gmail.com>

frenck committed:
Update home-assistant/wheels to 2023.09.1 (#100758)

* Update home-assistant/wheels to 2023.09.0

* Update home-assistant/wheels to 2023.09.1

jbouwh committed:
Correct some typo's in MQTT issue string (#100759)

rappenze committed:
Fibaro finish separation of scenes (#100734)

allenporter committed:
Add myself as a fitbit codeowner (#100766)

dknowles2 committed:
Bump pyschlage to 2023.9.1 (#100760)

joostlek committed:
Add media extractor tests (#100462)

* Add tests for media extractor

* Complete test coverage

* Fix test dep

allenporter committed:
Add tests for fitbit integration (#100765)

* Add tests for fitbit integration

* Update coveragerc

* Update test requirements

joostlek committed:
Call async added to hass super in Flo (#100453)

sdb9696 committed:
Bump ring-doorbell to 0.7.3 (#100688)

Bump ring to 0.7.3

lymanepp commented:
1) Tomorrow.io API has a limit of 55 fields per request. Will this change increase the number of queries required to fetch all fields?
1) Adding sensors that always contain "unknown" isn't useful. How many of the added fields are actually available for recently created free API tokens?
1) If/when this is merged, additional test cases are required.

joostlek committed:
Fix fitbit test code owner (#100772)

gjohansson-ST committed:
Bump pysensibo 1.0.35 (#100245)

* Bump pysensibo 1.0.34

* 1.0.35

* Mod tests

* revert refactoring

* Fix tests

gjohansson-ST committed:
Add strong to fan mode for Sensibo (#100773)

allenporter committed:
Fix rainbird unique id (#99704)

* Don't set a unique id for devices with no serial

* Add additional check for the same config entry host/port when there is no serial

* Update homeassistant/components/rainbird/config_flow.py

Co-authored-by: Robert Resch <robert@resch.dev>

* Update tests/components/rainbird/test_config_flow.py

Co-authored-by: Robert Resch <robert@resch.dev>

* Update tests/components/rainbird/test_config_flow.py

Co-authored-by: Robert Resch <robert@resch.dev>

---------

Co-authored-by: Robert Resch <robert@resch.dev>

gjohansson-ST committed:
Refactor Sensibo tests to use snapshot (#100775)

dknowles2 committed:
Add config flow to Hydrawise (#95589)

* Add config flow to Hydrawise

* Raise an issue when a YAML config is detected

* Add a test for YAML import

* Add missing __init__.py

* Update CODEOWNERS

* Update requirements_test_all.txt

* Add config flow data to strings.json

* Hande scan_interval not being in YAML on import

* Fix requirements

* Update deprecation dates

* Update requirements_test_all.txt

* Changes from review

* Update homeassistant/components/hydrawise/__init__.py

Co-authored-by: G Johansson <goran.johansson@shiftit.se>

* Add already_configured to strings.json

* Add back setup_platform functions

* Apply suggestions from code review

Co-authored-by: G Johansson <goran.johansson@shiftit.se>

* Add back setup_platform

* Update requirements_test_all.txt

* Run black on hydrawise/*.py

* Add missing import of HOMEASSISTANT_DOMAIN

* Use more specific errors in config flow

* Add additional tests

* Update config flow to use pydrawise.legacy

* Re-work YAML deprecation issues

* Revert some changes to binary_sensor, as requested in review

* Changes requested during review

* Apply suggestions from code review

Co-authored-by: G Johansson <goran.johansson@shiftit.se>

* Remove unused STE_USER_DATA_SCHEMA

Co-authored-by: G Johansson <goran.johansson@shiftit.se>

* Update comment in setup_platform

* Re-work the config flow again

* Apply suggestions from code review

Co-authored-by: G Johansson <goran.johansson@shiftit.se>

* Update tests

* Add back the _default_watering_timer attribute

* Bump deprecation dates

* Update requirements_test_all.txt

* Update CODEOWNERS

---------

Co-authored-by: G Johansson <goran.johansson@shiftit.se>
Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

ntilley905 committed:
Bump faadelays to 2023.8.0 (#100700)

* Update component to use new API version

* Revert new features, implement #95546, bump library

* Revert #95546 changes, remove NOTAM

allenporter committed:
Add more test coverage for fitbit sensors (#100776)

ntilley905 committed:
Clean up FAA Delays constants (#100788)

Move const to platform

AtomBrake committed:
Update powerwall password description (#100389)

Update strings.json

Updated wording of how to find password on newer model gateways

MartinHjelmare commented:
Please move the weather platform changes to a separate PR.

Jc2k committed:
Expose bluetooth availability tracking interval controls to integrations (#100774)

jbouwh committed:
Avoid redundant calls to async_write_ha_state in mqtt device_tracker (#100767)

Avoid redundant calls to async_ha_write_state

frenck committed:
Update home-assistant/builder to 2023.09.0 (#100797)

emontnemery committed:
Remove support for excluding attributes in recorder platforms (#100679)

chemelli74 committed:
Fix Comelit device info (#100587)

dirrgang commented:
> 1. Tomorrow.io API has a limit of 55 fields per request. Will this change increase the number of queries required to fetch all fields?
> 
>     2. Adding sensors that always contain "unknown" isn't useful. How many of the added fields are actually available for recently created free API tokens?
> 
>     3. If/when this is merged, additional test cases are required.

1. It will increase the number of queries. Perhaps adding some logic that only queries the API for sensors that are actually enabled would be a good idea.
2. I can't find it in the documentation right now, but I'm 99% sure that they explicitly stated that all core data layer fields are part of the free access tier. Something like "centimetres of snow" will remain unknown until there is actually any snow, for example. The fields in general are location dependent. 
3. Alright, I'll look into that.

lymanepp commented:
1) Free API tokens have a limit on requests per day. Requiring more calls will affect that.
1) Your screenshots have many unknown values. Do they ever have a value?

dirrgang commented:
> 1. Free API tokens have a limit on requests per day. Requiring more calls will affect that.
> 
>     2. Your screenshots have many unknown values. Do they ever have a value?

1. Oh, I misunderstood you then. No, there's no extra calls. The same number of calls, just with more attributes.
2. Probably. I'd have to test it - the fields are location dependent. It's late summer here, so we have no snow, thus the snow sensors don't show anything.

scolby33 committed:
Allow workday sensor to be configured without a country (#93048)

* Merge branch 'dev' into workday_without_country

* ruff

* remove province check

* Remove not needed test

* Mod config flow

---------

Co-authored-by: G Johansson <goran.johansson@shiftit.se>

lymanepp commented:
1) Enable debugging and see how many requests are actually made. See my earlier #1 content to understand why I asked this.
1) Try different locations around the world to see if they're ever populated. Definitely remove values that are always unknown.

Jc2k committed:
Preserve private ble device broadcast interval when MAC address rotates (#100796)

bdraco committed:
Bump bluetooth-data-tools to 0.12.0 (#100794)

Noltari committed:
Update AEMET-OpenData to v0.4.5 (#100818)

Signed-off-by: Álvaro Fernández Rojas <noltari@gmail.com>

allenporter committed:
Rewrite fitbit sensor API response value parsing (#100782)

* Cleanup fitbit sensor API parsing

* Remove API code that is not used yet

* Remove dead code for battery levels

Small API parsing cleanup

* Address PR feedback

* Update homeassistant/components/fitbit/sensor.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

---------

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

natekspencer committed:
Bump pylitterbot to 2023.4.8 (#100811)

trnila committed:
Add Mysensors battery sensor (#100749)

* Move child related stuff to MySensorsChildEntity

* Dispatch signal for newly discovered MySensors node

* Create battery entity for each MySensors node

* Removed ATTR_BATTERY_LEVEL attribute from each node sensor

Attribute is redundant with newly introduced battery sensor entity

* Apply suggestions from code review

---------

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

bachya committed:
Add missing SimpliSafe binary sensors (#100820)

dirrgang committed:
Revert "Added all core sensors and extended forecast."

This reverts commit a8a456ea98160202f5f7949a16cccf4d3b4218a6.

dirrgang committed:
Revert "Added all core sensors and extended forecast."

This reverts commit a8a456ea98160202f5f7949a16cccf4d3b4218a6.

dirrgang committed:
Added all sensors of core data plane.

dirrgang committed:
Added all sensors of core data plane.

bdraco committed:
Bump aioesphomeapi to 16.0.6 (#100826)

changelog: https://github.com/esphome/aioesphomeapi/compare/v16.0.5...v16.0.6

gjohansson-ST committed:
Change duration for timer.start service to only change running duration (#99628)

* Get back duration for timer

* running duration

* Mods

* Finish

* Fix start call

* remove restore idle

* running duration not None

* fix tests

gjohansson-ST committed:
Create repairs in Workday if country or province is wrong (#98753)

* Repairs workday

* fix if not province exist

* Tests repairs

* Add tests

* Finalize tests

* Fix feedback

* simplify

* Less translations

edenhaus reviewed (changes requested):
Can you please rebase and fix the merge conflicts?

After that, I can start reviewing your PR

dependabot[bot] committed:
Bump actions/checkout from 4.0.0 to 4.1.0 (#100836)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dirrgang commented:
I did some more testing and removed sensors which were perpetually in a state of *unknown*.

However, this raises the question of what to do with already existing sensors, which share the same fate.

The air quality sensors and global horizontal irradiance have always been unknown as well, do those work for anyone? Otherwise, they might be candidates for removal also. 

Current sensors:

![grafik](https://github.com/home-assistant/core/assets/29854606/3eb85519-1530-4828-91c9-4550f42a7182)
![grafik](https://github.com/home-assistant/core/assets/29854606/7a709334-0ad5-4b6b-b2d9-599e5d24dd70)

dirrgang committed:
Removed sensors without output
(and added wind speed)

dirrgang committed:
Removed sensors without output
(and added wind speed)

dirrgang committed:
Cleaned up init and added icons

dirrgang committed:
Cleaned up init and added icons

cereal2nd committed:
Add duotecno climate (#99333)

* Add duotecno climate

* Add climate to .coveragerc

* Update homeassistant/components/duotecno/climate.py

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* Update homeassistant/components/duotecno/climate.py

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* more comments

* more comments

* more comments

* more comments

* fix typo

* Add translation key

---------

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

alexyao2015 committed:
Address late review on Life360 button (#100740)

dieselrabbit committed:
Explicitly define ScreenLogic entity descriptions (#100173)

cereal2nd committed:
Add binary sensors for duotecno (#100844)

* Add binary sensors for duotecno

* Add comments

balloob committed:
Allow passing a wake word ID to detect wake word (#100832)

* Allow passing a wake word ID to detect wake word

* Do not inject default wake words in wake_word integration

Jc2k committed:
Add a select entity for homekit temperature display units (#100853)

emontnemery committed:
Store wakeword settings in assist pipelines (#100847)

* Store wakeword settings in assist pipelines

* wakeword -> wake_word

* Remove unneeded variable

cdce8p committed:
Update pylint to 2.17.6 (#100849)

Jc2k committed:
Add valve position sensor for Eve Thermo (#100856)

bdraco committed:
Provide a better model for HomeKit service entries (#100848)

bdraco committed:
Add support for receivers to HomeKit (#100717)

Jc2k committed:
Add duration and sensitivity configuration for Eve Motion (#100861)

emontnemery committed:
Rename wake_word.async_default_engine to wake_word.async_default_entity (#100855)

* Rename wake_word.async_default_engine to wake_word.async_default_entity

* tweak

* Some more rename

* Update tests

elmurato committed:
Simplify Minecraft Server SRV handling (#100726)

jbouwh committed:
Avoid redundant calls to async_write_ha_state for mqtt fan (#100777)

Avoid redundant calls to async_write_ha_state

jbouwh committed:
Avoid redundant calls to async_write_ha_state in mqtt humidifier (#100781)

Avoid redundant calls to async_write_ha_state

jbouwh committed:
Rework and added tests for mqtt event (#100769)

Use write_state_on_attr_change and add tests

Jc2k committed:
Fix missing device class on Velux Windows (#100863)

jbouwh committed:
Avoid redundant calls to async_write_ha_state in mqtt lawn_mower (#100795)

Avoid redundant calls to async_write_ha_state

jbouwh committed:
Avoid redundant calls to async_write_ha_state in mqtt lock (#100802)

Avoid redundant calls to async_write_ha_state

jbouwh committed:
Avoid redundant calls to async_write_ha_state in mqtt number (#100808)

Avoid redundant calls to async_write_ha_state

jbouwh committed:
Avoid redundant calls to async_write_ha_state in mqtt select (#100809)

Avoid redundant calls to async_write_ha_state

jbouwh committed:
Avoid redundant calls to async_ha_write_state in mqtt switch (#100815)

Avoid redundant calls to async_ha_write_state

jbouwh committed:
Avoid redundant calls to async_ha_write_state in mqtt text (#100816)

Avoid redundant calls to async_ha_write_state

jbouwh committed:
Avoid redundant calls to async_ha_write_state mqtt update platform (#100819)

Avoid redundant calls to async_ha_write_state

cdce8p committed:
Update types packages (#100850)

emontnemery committed:
Use wake word settings in assist pipeline runs (#100864)

jbouwh committed:
Rework and fix mqtt siren writing state and attributes (#100871)

Rework mqtt siren writing state and attributes

jpbede committed:
Handle json decode exception in co2signal (#100857)

* Handle json decode exception in co2signal

* Update homeassistant/components/co2signal/coordinator.py

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* Fix import

---------

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

bdraco committed:
Bump led-ble to 1.0.1 (#100873)

jbouwh committed:
Avoid redundant calls to async_write_ha_state in mqtt vacuum (#100799)

* Avoid redundant calls to async_write_ha_state

* Add comment

* Rephrase

jbouwh committed:
Add tests for mqtt image (#100793)

* Rework mqtt image writing state

* Revert mixin changes, add attr

* Revert code changes

jpbede committed:
Parametrize more co2signal config flow tests (#100882)

* Clean up co2signal tests

* Some more clean up

* Use named parameter ids of parametrize

jbouwh committed:
Avoid redundant calls to async_write_ha_state in mqtt siren (#100813)

* Avoid redundant calls to async_write_ha_state

* Add comment

jbouwh committed:
Add type hints for intent_script integration (#99393)

* Add type hints for intent_script integration

* Correct 2nd typo

* omit total=False as all options are set

joostlek committed:
Bump yt-dlp to 2023.9.24 (#100884)

bdraco committed:
Bump dbus-fast to 2.10.0 (#100879)

bdraco committed:
Bump zeroconf to 0.114.0 (#100880)

bdraco committed:
Bump async-upnp-client to 0.36.0 (#100881)

jbouwh committed:
Add doorbell event to google_assistant (#97123)

* First attempt async_report_state_all

* Move notificationSupportedByAgent to SYNC response

* Make notificationSupportedByAgent conditional

* Add generic sync_options method

* Report event

* Add event_type as ID

* User UUID, imlement query_notifications

* Refactor query_notifications

* Add test

* MyPy

* Unreachable code

* Tweak

* Correct notification message

* Timestamp was wrong unit, it should be in seconds
* Can only allow doorbell class, since it's the only type

* Fix test

* Remove unrelated changes - improve coverage

* Additional tests

---------

Co-authored-by: Joakim Plate <elupus@ecce.se>

Jc2k committed:
Simplify homekit_controller tests with snapshots (#100885)

Jc2k committed:
Bump aiohomekit to 3.0.5 (#100886)

jbouwh committed:
Make sure time is changed in mqtt event test (#100889)

synesthesiam committed:
Use webrtc-noise-gain for audio enhancement in Assist pipelines (#100698)

* Use webrtc-noise-gain instead of webrtcvad package

* Switching to ProcessedAudioChunk

* Refactor VAD and fix tests

* Add vad no chunking test

* Add test that runs audio enhancements

allenporter committed:
Simplify fitbit unit system and conversions (#100825)

* Simplify fitbit unit conversions

* Use enum values in unit system schema

* Use fitbit unit system enums

allenporter committed:
Add Rain Bird irrigation calendar (#87604)

* Initial version of a calendar for the rainbird integration

* Improve calendar support

* Revert changes to test fixtures

* Address ruff error

* Fix background task scheduling

* Use pytest.mark.freezetime to move to test setup

* Address PR feedback

* Make refresh a member

* Merge rainbird and calendar changes

* Increase test coverage

* Readability improvements

* Simplify timezone handling

balloob committed:
Allow fetching wake word entity info (#100893)

jpbede committed:
Enable strict typing in london underground (#100563)

* Enable strict typing in london underground

* Change typing from Any

* Remove redundant cast

* Change from Mapping to dict

gjohansson-ST committed:
Add date range to Workday (#96255)

dknowles2 committed:
Add device info to Hydrawise (#100828)

* Add device info to Hydrawise

* Apply suggestions from code review

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* Remove _attr_has_entity_name

---------

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

joostlek committed:
Add coordinator to Withings (#100378)

* Add coordinator to Withings

* Add coordinator to Withings

* Fix tests

* Remove common files

* Fix tests

* Fix tests

* Rename to Entity

* Fix

* Rename webhook handler

* Fix

* Fix external url

* Update homeassistant/components/withings/entity.py

Co-authored-by: Luke Lashley <conway220@gmail.com>

* Update homeassistant/components/withings/entity.py

Co-authored-by: Luke Lashley <conway220@gmail.com>

* Update homeassistant/components/withings/entity.py

Co-authored-by: Luke Lashley <conway220@gmail.com>

* Update homeassistant/components/withings/entity.py

Co-authored-by: Luke Lashley <conway220@gmail.com>

* fix imports

* Simplify

* Simplify

* Fix feedback

* Test if this makes changes clearer

* Test if this makes changes clearer

* Fix tests

* Remove name

* Fix feedback

---------

Co-authored-by: Luke Lashley <conway220@gmail.com>

cereal2nd committed:
Bump pyduotecno to 2023.9.0 (#100900)

joostlek committed:
Rename Withings coordinator file (#100899)

Rename common.py to coordinator.py

emontnemery committed:
Rename PipelineData.pipeline_runs to pipeline_debug (#100907)

joostlek committed:
Add entity translations to Wallbox (#99021)

Co-authored-by: Robert Resch <robert@resch.dev>

cdce8p committed:
Update cryptography to 41.0.4 (#100911)

bdraco committed:
Mark Bluetooth scanner as not scanning when watchdog timeout is reached (#100738)

dirrgang committed:
Add state classes to Tomorrowio sensors (#100692)

emontnemery committed:
Remove duplicated call to `PipelineRun.end` from `PipelineInput.execute` (#100909)

Remove duplicated call to PipelineRun.end from PipelineInput.execute

cdce8p committed:
Remove async-timeout as core dependency (#100912)

joostlek committed:
Cleanup Withings const import (#100914)

steffenrapp committed:
Add missing input_button service translation (#100387)

Noltari committed:
Bump aioairzone-cloud to 0.2.2 (#100915)

Signed-off-by: Álvaro Fernández Rojas <noltari@gmail.com>

dknowles2 committed:
Make Hydrawise compliant with new naming standards (#100921)

* Make Hydrawise compliant with new naming standards

* Update binary_sensor.py

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

---------

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

joostlek committed:
Add entity translations to SRP Energy (#99011)

engrbm87 committed:
Handle authorization error in glances config flow (#100866)

* Handle authroization error in glances config flow

* Remove validate_input method and expections

* update tests

emontnemery committed:
Tweak pipeline.multiply_volume (#100905)

allenporter committed:
Update fitbit client to use asyncio (#100933)

bdraco committed:
Small cleanup to isy994 extra_state_attributes (#100935)

bdraco committed:
Cache the latest short term stat id for each metadata_id on each run (#100535)

bdraco committed:
Bump zeroconf to 0.115.0 (#100931)

emontnemery committed:
Tweak test wake_word.test_init.test_detected_entity (#100910)

engrbm87 committed:
Add re-auth flow to glances integration (#100929)

* Add reauth flow to glances integration.

* add reauth string

* add reauth strings

joostlek committed:
Add body_exists to MockRequest in aiohttp util (#100932)

* Add body_exists to MockRequest in aiohttp util

* Add body_exists to MockRequest in aiohttp util

* Add body_exists to MockRequest in aiohttp util

bramkragten committed:
Update frontend manifest for new icons (#100936)

mib1185 committed:
Add support to remove orphan devices in AVM FRITZ!SmartHome (#100739)

joostlek committed:
Add config flow to AfterShip (#100872)

* Add config flow to Aftership

* Add config flow to Aftership

* Fix schema

* Update homeassistant/components/aftership/strings.json

Co-authored-by: Robert Resch <robert@resch.dev>

* Fix feedback

---------

Co-authored-by: Robert Resch <robert@resch.dev>

emontnemery committed:
Rename WakeWord.ww_id to WakeWord.id (#100903)

* Rename WakeWord.ww_id to WakeWord.wake_word_id

* Revert unrelated changes

* Rename to id

* Correct rebase

jbouwh committed:
Intialize mqtt lock in an unknown state in pessimistic mode (#100943)

Intialize mqtt lock as unknown in pessimistic mode

joostlek committed:
Remove setting name in AnthemAV config flow (#99148)

joostlek committed:
Use automatic title during config flow setup in Aurora (#99199)

joostlek committed:
Call async added to hass super in Smart Meter Texas (#100445)

joostlek committed:
Call async added to hass super in Risco (#100444)

jpbede committed:
Use snapshot assertion for wiz diagnostics test (#99154)

joostlek committed:
Call async added to hass super in Airvisual (#100449)

Olen committed:
Move fetching of sw_version for Twinkly (#100286)

jbouwh committed:
Remove redundant initial assigment for mqtt siren (#100945)

chemelli74 committed:
Improve Vodafone Station setup dialog messages (#100937)

jbouwh committed:
Remove unused attribute for MQTT lawn_mower (#100946)

Remove unused attributes for MQTT lawn_mower

emontnemery committed:
Abort wake word detection when assist pipeline is modified (#100918)

chemelli74 committed:
Bump aiovodafone to 0.3.1 (#100944)

elafargue committed:
Add Medcom Bluetooth integration (#100289)

Co-authored-by: J. Nick Koston <nick@koston.org>

emontnemery committed:
Simplify wake_word/info + improve test coverage (#100902)

* Simplify wake_word/info + improve test coverage

* Fix test

* Revert unrelated changes

dirrgang committed:
Merge branch 'dev' into tomorrowio-sensors

dirrgang committed:
Rebase and state/device class for all sensors

dirrgang commented:
Well, that didn't bloody work.
